File: composeApp/src/commonMain/kotlin/com/antigravity/webapp/ui/components/Sidebar.kt
package com.antigravity.webapp.ui.components

import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.material3.*
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.unit.dp
import org.jetbrains.compose.ui.tooling.preview.Preview

data class MenuItem(
    val route: String,
    val label: String,
    val icon: ImageVector,
    val roles: List<String>
)

val MainMenuItems = listOf(
    MenuItem("home", "Dashboard", Icons.Default.Home, listOf("ADMIN", "OPERARIO", "VISOR")),
    MenuItem("travels", "Gestión de Viajes", Icons.Default.LocalShipping, listOf("ADMIN", "OPERARIO", "VISOR")),
    MenuItem("master-data", "Datos Maestros", Icons.Default.Analytics, listOf("ADMIN")),
    MenuItem("logistics-liquidations", "Liquidación Choferes", Icons.Default.Payments, listOf("ADMIN", "OPERARIO")),
    MenuItem("logistics-fuel", "Rendimiento Combustible", Icons.Default.LocalGasStation, listOf("ADMIN", "OPERARIO")),
    MenuItem("consumption", "Despacho y Stock", Icons.Default.Inventory, listOf("ADMIN", "OPERARIO")),
    MenuItem("simulator", "Simulador de Costos", Icons.Default.Calculate, listOf("ADMIN", "OPERARIO", "VISOR")),
    MenuItem("price-share", "Placa de Precios", Icons.Default.PriceChange, listOf("ADMIN", "OPERARIO")),
    MenuItem("checks", "Gestión de Cheques", Icons.Default.AccountBalanceWallet, listOf("ADMIN", "OPERARIO")),
    MenuItem("accounting", "Caja General", Icons.Default.AccountBalance, listOf("ADMIN", "OPERARIO")),
    MenuItem("frigorifico", "Caja Frigorífico", Icons.Default.Store, listOf("ADMIN", "OPERARIO")),
    MenuItem("clients", "Clientes y Cuentas", Icons.Default.People, listOf("ADMIN")),
    MenuItem("establishments", "Sucursales y Personal", Icons.Default.Store, listOf("ADMIN")),
    MenuItem("settings", "Configuración", Icons.Default.Settings, listOf("ADMIN")),
    MenuItem("contact", "Info y Contacto", Icons.Default.Contacts, listOf("ADMIN", "OPERARIO", "VISOR"))
)

@Composable
fun Sidebar(
    currentRoute: String?,
    onNavigate: (String) -> Unit,
    onLogout: () -> Unit,
    userRole: String = "ADMIN",
    modifier: Modifier = Modifier
) {
    val visibleItems = MainMenuItems.filter { it.roles.contains(userRole) }

    Column(
        modifier = modifier
            .width(280.dp)
            .fillMaxHeight()
            .padding(12.dp)
    ) {
        Text(
            text = "Antigravity",
            style = MaterialTheme.typography.titleLarge,
            modifier = Modifier.padding(vertical = 24.dp, horizontal = 16.dp)
        )

        LazyColumn(
            modifier = Modifier.weight(1f)
        ) {
            items(visibleItems) { item ->
                NavigationDrawerItem(
                    label = { Text(item.label) },
                    icon = { Icon(item.icon, contentDescription = null) },
                    selected = currentRoute == item.route,
                    onClick = { onNavigate(item.route) },
                    modifier = Modifier.padding(NavigationDrawerItemDefaults.ItemPadding)
                )
            }
        }

        Spacer(modifier = Modifier.weight(0.05f))

        // Botón de Logout
        NavigationDrawerItem(
            label = { 
                Text(
                    text = "Cerrar Sesión", 
                    color = MaterialTheme.colorScheme.error 
                ) 
            },
            icon = { 
                Icon(
                    imageVector = Icons.Default.ExitToApp, 
                    contentDescription = null,
                    tint = MaterialTheme.colorScheme.error
                ) 
            },
            selected = false,
            onClick = onLogout,
            modifier = Modifier.padding(NavigationDrawerItemDefaults.ItemPadding)
        )
    }
}

@Preview
@Composable
private fun SidebarPreview() {
    MaterialTheme {
        Sidebar(
            currentRoute = "home",
            onNavigate = {},
            onLogout = {}
        )
    }
}

File: src/frameworks/ui/components/Sidebar.js
import { el } from '../../../frameworks/utils/dom.js';

/**
 * Renders the navigation sidebar with role-based visibility and premium logical groupings.
 * Organizes 15 separate system tools into distinct operational, financial, analytical,
 * and administrative areas for enhanced ergonomics and premium visuals.
 *
 * @param {HTMLElement} container - The container where the menu list will be rendered.
 * @param {Function} onNavigate - Callback triggered when a menu link is clicked.
 * @param {string} userRole - Current user's role authorization (ADMIN, OPERARIO, VISOR).
 */
export function renderSidebar(container, onNavigate, userRole) {
  if (!container) return;

  const currentRole = userRole || 'VISOR';

  // --- curating navigation groups & access authorization ---
  const groups = [
    {
      title: 'OPERACIONES & LOGÍSTICA',
      items: [
        { id: 'dashboard', label: '📊 Dashboard', roles: ['ADMIN', 'OPERARIO', 'VISOR'] },
        { id: 'travels', label: '🚛 Gestión de Viajes', roles: ['ADMIN', 'OPERARIO', 'VISOR'] },
        { id: 'logistics-liquidations', label: '💵 Liquidación Choferes', roles: ['ADMIN', 'OPERARIO'] },
        { id: 'logistics-fuel', label: '⛽ Rendimiento Combustible', roles: ['ADMIN', 'OPERARIO'] },
        { id: 'consumption', label: '🥩 Despacho y Stock', roles: ['ADMIN', 'OPERARIO'] }
      ]
    },
    {
      title: 'FINANZAS & CRÉDITO',
      items: [
        { id: 'checks', label: '💸 Gestión de Cheques', roles: ['ADMIN', 'OPERARIO'] },
        { id: 'accounting', label: '💰 Caja General', roles: ['ADMIN', 'OPERARIO'] },
        { id: 'frigorifico', label: '🏢 Caja Frigorífico', roles: ['ADMIN', 'OPERARIO'] }
      ]
    },
    {
      title: 'HERRAMIENTAS & ANÁLISIS',
      items: [
        { id: 'simulator', label: '🧮 Simulador de Costos', roles: ['ADMIN', 'OPERARIO', 'VISOR'] },
        { id: 'price-share', label: '📲 Placa de Precios', roles: ['ADMIN', 'OPERARIO'] }
      ]
    },
    {
      title: 'SISTEMA & CONFIGURACIÓN',
      items: [
        { id: 'master-data', label: '⚙️ Datos Maestros', roles: ['ADMIN'] },
        { id: 'clients', label: '👥 Clientes y Cuentas', roles: ['ADMIN'] },
        { id: 'establishments', label: '🏢 Sucursales y Personal', roles: ['ADMIN'] },
        { id: 'settings', label: '⚙️ Configuración', roles: ['ADMIN'] },
        { id: 'contact', label: '📖 Info y Contacto', roles: ['ADMIN', 'OPERARIO', 'VISOR'] }
      ]
    }
  ];

  container.innerHTML = '';
  const ul = el('ul', { style: 'list-style: none; padding: 0.5rem; margin: 0; display: flex; flex-direction: column; gap: 0.25rem;' });

  let isFirstGroup = true;

  groups.forEach(group => {
    // Filter visible items in this group based on current user role permissions
    const visibleItems = group.items.filter(item => item.roles.includes(currentRole));

    if (visibleItems.length === 0) return;

    // Render group title divider
    const groupHeader = el('div', {
      text: group.title,
      style: `font-size: 0.68rem; font-weight: 800; color: var(--text-muted); opacity: 0.45; letter-spacing: 1.2px; padding: ${isFirstGroup ? '0.5rem' : '1.25rem'} 0.75rem 0.45rem 0.75rem; text-transform: uppercase; ${isFirstGroup ? '' : 'border-top: 1px solid rgba(255,255,255,0.03);'}`
    });
    ul.appendChild(groupHeader);
    isFirstGroup = false;

    // Render each item
    visibleItems.forEach(item => {
      const li = el('li', {
        text: item.label,
        classes: ['nav-item'],
        attrs: { 'data-view': item.id },
        style: 'padding: 0.75rem 1rem; border-radius: 12px; cursor: pointer; transition: all 0.22s cubic-bezier(0.4, 0, 0.2, 1); margin-bottom: 0.15rem; font-weight: 600;'
      });

      li.onclick = () => {
        // Toggle active status cleanly
        ul.querySelectorAll('li').forEach(l => l.classList.remove('active'));
        li.classList.add('active');

        if (typeof onNavigate === 'function') {
          onNavigate(item.id);
        }

        // Close sidebar drawer automatically on responsive screens/mobile viewport sizes
        document.body.classList.remove('sidebar-open');
      };

      ul.appendChild(li);
    });
  });

  // Render bottom premium logout action button
  const logoutLi = el('li', {
    text: '🚪 Cerrar Sesión',
    classes: ['nav-item', 'logout-item'],
    style: 'padding: 0.75rem 1rem; margin-top: 2rem; border-radius: 12px; cursor: pointer; color: var(--danger); font-weight: 700; transition: all 0.22s ease;'
  });

  logoutLi.onclick = () => {
    if (confirm('¿Deseas cerrar sesión del sistema?')) {
      window.dispatchEvent(new CustomEvent('app:logout'));
    }
  };

  ul.appendChild(logoutLi);
  container.appendChild(ul);
}
